File: LeetCode/487-max-consecutive-ones-ii/ex_487.java
public class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int maxCount = 0, left = 0, right = 0;
        for (int n : nums) {
            right++;
            if (n == 0) {
                left = right;
                right = 0;
            }
            maxCount = Math.max(maxCount, left + right);
        }
        return maxCount;
    }
}

File: LeetCode/487-max-consecutive-ones-ii/ex_487.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int max_count = 0, left = 0, right = 0;
        for (auto &n : nums) {
            ++right;
            if (n == 0) {
                left = right;
                right = 0;
            }
            max_count = max(max_count, left + right);
        }
        return max_count;
    }
};